lib/client.js

```diff
@@ -124,7 +124,11 @@ function isValidEvent(client, event) {
 function triggerEvent(client, eventName, data) {
   if (!client._messageHandlers[eventName]) { return false; }
   client._messageHandlers[eventName].forEach(function(handler) {
-    handler(data);
+    if (Array.isArray(data)) {
```

Comment:
What is data was being sent as an array already? Could we send this from the framework using a special key to distinguish it further, like `responseArgs`?

e.g.
```
if (data.eventArgs) {
  handler.apply(undefined, data.eventArgs);
}
```

Reply:
This can only happen to `zaf.` events. And I looked into framework code, this only happens for non-ajax requests, so if an ajax request returns an array, it won't hit this code path.

Do you know of any events that have an array as the first argument?

Reply:
Actually, we already do this strange thing, where we wrap ajax data into `{ responseArgs: data}`, and then in the sdk, we unwrap it. Why is this?

Reply:
Yes, that's what I meant. It's actually legacy from v1. But I don't think it's weird. It makes the intent more explicit.

Reply:
The intent of passing arguments to the app's event handler, rather than any other kind of data that may be internal to the framework.

Reply:
I don't see how the intent is wrong. The behaviour you described is the same I described: "passing arguments to the app's event handler", no? It's just not explicit. I can't tell just by reading this code what data is used for. 

As for over-engineering, I would agree with you if this was all one project in one repo. It would be easy to change the interface and consumers at the same time. This not being the case, I prefer to nest it so that if we need to pass additional internal properties it's easy to do without introducing temporary backwards compatibility code.